setup/send_assets_to_multi_giveaway_1.ts

```diff
@@ -1,141 +1,74 @@
-import fs from 'fs';
-import hre, {getNamedAccounts} from 'hardhat';
+/**
+ * How to use:
+ *  - yarn execute <NETWORK> ./setup/send_assets_to_multi_giveaway_1.ts <GIVEAWAY_NAME>
+ *
+ * GIVEAWAY_NAME: from data/giveaways/multi_giveaway_1/detective_letty.json then the giveaway name is: detective_letty
+ */
+import fs from 'fs-extra';
+import hre from 'hardhat';
+import {BigNumber} from '@ethersproject/bignumber';
 import {DeployFunction} from 'hardhat-deploy/types';
-import {MultiClaim} from '../lib/merkleTreeHelper';
+import {MultiClaim, AssetHash} from '../lib/merkleTreeHelper';
+const {deployments} = hre;
+const {execute, catchUnknownSigner, read} = deployments;
+const networkName = hre.network.name;
+
+const args = process.argv.slice(2);
+const claimFile = args[0];
+
+function getAssets(giveawayName: string): AssetHash {
+  const path = `./data/giveaways/multi_giveaway_1/${giveawayName}.json`;
+  const json: Array<MultiClaim> = fs.readJSONSync(path);
+  const assetIdsCount: AssetHash = {};
+  json.forEach((claim) => {
+    claim.erc1155.forEach(({ids, values}) => {
+      ids.forEach((id, index) => {
+        if (!assetIdsCount[id]) assetIdsCount[id] = 0;
+        assetIdsCount[id] += values[index];
+      });
+    });
+  });
+  return assetIdsCount;
+}
 
 const func: DeployFunction = async function () {
-  const {deployments} = hre;
-  const {execute, catchUnknownSigner} = deployments;
-
-  const {deployer} = await getNamedAccounts();
-
-  let owner;
-  let CLAIM_FILE;
-  let CONFIG_FILE;
-
-  switch (hre.network.name) {
-    case 'mainnet':
-      owner = '';
-      CLAIM_FILE = 'data/giveaways/multi_giveaway_1/claims_0_mainnet.json';
-      CONFIG_FILE = 'data/giveaways/multi_giveaway_1/config_mainnet.ts';
-      break;
-    case 'rinkeby':
-      owner = deployer;
-      CLAIM_FILE = 'data/giveaways/multi_giveaway_1/claims_0_rinkeby.json';
-      CONFIG_FILE = 'data/giveaways/multi_giveaway_1/config_rinkeby.ts';
-      break;
-    default:
-      owner = deployer;
-      CLAIM_FILE = 'data/giveaways/multi_giveaway_1/claims_0_hardhat.json';
-      CONFIG_FILE = 'data/giveaways/multi_giveaway_1/config_hardhat.ts';
-  }
-
-  if (!owner || owner === '') {
-    return;
-  }
-
+  const assetIdsCount = await getAssets(claimFile);
+  const owner =
+    networkName === 'mainnet'
+      ? '0x7A9fe22691c811ea339D9B73150e6911a5343DcA'
+      : '0x5BC3D5A39a50BE2348b9C529f81aE79f00945897';
```

Comment:
Is there a way to not hardcode the addresses ?

Reply:
Changed to sandboxAccount